cli/src/etosctl/plugin/plugin.py

```diff
@@ -0,0 +1,52 @@
+# Copyright Axis Communications AB.
+#
+# For a full list of individual contributors, please see the commit history.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Etosctl plugin.
+
+ETOS will load all plugins that register themselves to the Plugins list.
+
+>>> from etosctl.plugin import register
+>>> register("myplugin", TestPlugin)
+"""
+from typing import Optional
+
+PLUGINS = {}
+
+
+class Plugin:
```

Comment:
We could get the class `Plugin` to define a `metaclass` that automatically registers itself as a plugin, instead of having plugin creators register them, themselves.

```
class PluginRegistry(type):
    plugins: List = list()

    def __init__(cls, name, bases, attrs):
        super().__init__(cls)
        PluginRegistry.plugins.append(cls)


class Plugin(object, metaclass=PluginRegistry):
    ...
```